src/protocol/serializers.ts

```diff
@@ -151,5 +151,5 @@ function isDate(obj: any): obj is Date {
 }
 
 function isError(obj: any): obj is Error {
-  return obj instanceof Error || (obj && obj.__proto__ && obj.__proto__.name === 'Error');
+  return obj instanceof Error || obj && obj.__proto__.name === 'Error' || obj && isError(obj.__proto__);
```

Comment:
Was missing a check for `obj.__proto__` truthiness. People can throw an object with `{ __proto__: null }`.

```suggestion
  return (obj instanceof Error) || (obj && obj.__proto__ && obj.__proto__.name === 'Error') || (obj && isError(obj.__proto__));
```

Reply:
clarified offline: optional chaining solution is fine.